WIP: Documentation for users & permissions

kgodey opened the pull request:
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, open one before creating this pull request. -->
Fixes #2455 

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. --> 
- [ ] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [ ] My pull request targets the `master` branch of the repository
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

rajatvijay commented on docs/docs/product/users.md:
```diff
@@ -0,0 +1,129 @@
+5. Click on the `Delete User` button.
+6. In the confirmation modal, again click on the `Delete User` button.
+
+## Granting database-level permissions
```
@kgodey do we want to add steps for editing permissions too? As of now UI doesn't support that feature, the user will have to delete the current permissions and grant the new permission.

kgodey replied:
No, it's fine, we can mention that the user will need to delete current permissions and add a new permission.

kgodey commented:
@rajatvijay I pushed some changes.
- I removed references that were very specific to the current UI. This seemed unnecessary (our UI is pretty intuitive) and also I think it'll be a pain to maintain as we change UI.
- I used MkDocs admonitions for better styling.
- I removed usage of `backticks` in favor of **bold**, I think `backticks` should be reserved for code.
- I restructured the docs a bit.

I left some *TODO* sections. Please fill them out!

rajatvijay commented:
@kgodey I have filled all of the TODOs and also added allowed actions per permission for explorations.

kgodey commented:
Thanks @rajatvijay I added a line about data explorer usage, I'm merging now.
